LightBlue.WorkerService/Program.cs

```diff
@@ -18,16 +25,127 @@ static int Main(string[] args)
                 x.Service<WorkerHost>(service =>
                 {
                     service.ConstructUsing(s => new WorkerHost(settings));
-                    service.WhenStarted(s => s.Start());
+                    service.WhenStarted((s, hc) => s.Start(hc));
                     service.WhenStopped(s => s.Stop());
                 });
 
                 x.RunAsLocalSystem();
                 x.SetDescription(string.Format("LightBlue {0} WorkerRole Windows Service", settings.ServiceTitle));
                 x.SetDisplayName(settings.ServiceTitle + " Service");
                 x.SetServiceName(settings.ServiceTitle);
-                x.StartManually();
+                x.StartAutomatically();
+                x.EnableShutdown();
+                x.EnableServiceRecovery(rc =>
+                {
+                    rc.RestartService(1);
+                });
             });
         }
+
+        public class WorkerHost
```

Comment:
It looks like we've moved this over to a different assembly, but perhaps it should still live in a separate file?